holoviews/plotting/bokeh/plot.py

```diff
@@ -586,13 +586,13 @@ def initialize_plot(self, ranges=None, plots=[]):
                 passed_plots.append(None)
 
         plot = gridplot(plots[::-1],
-                        merge_tools=self.merge_tools,
+                        merge_tools=False,
                         sizing_mode=self.sizing_mode,
                         toolbar_location=self.toolbar)
         if self.sync_legends:
             sync_legends(plot)
         plot = self._make_axes(plot)
-        if bokeh3 and hasattr(plot, "toolbar"):
+        if bokeh3 and hasattr(plot, "toolbar") and self.merge_tools:
```

Comment:
Does that mean this was previously incorrect? Also what happens in the bokeh2 case now?

Reply:
I think so. 

I have ignored Bokeh 2 as I plan to remove it soon with https://github.com/holoviz/holoviews/pull/5891. We don't even test for it anymore after the ImageStack PR was merged.

Reply:
Okay, sounds good.